concordium-consensus/src/Concordium/KonsensusV1/Consensus.hs

```diff
@@ -31,4 +28,87 @@ class MonadMulticast m where
 newtype BakerContext = BakerContext
     { _bakerIdentity :: BakerIdentity
     }
-makeClassy ''BakerContext
\ No newline at end of file
+
+makeClassy ''BakerContext
+
+-- |A Monad for timer related actions.
+class MonadTimeout m where
+    -- |Reset the timeout from the supplied 'Duration'.
+    resetTimer :: Duration -> m ()
+
+-- |Return 'Just FinalizerInfo' if the consensus running
+-- is part of the of the provided 'BakersAndFinalizers'.
+-- Otherwise return 'Nothing'.
+isBakerFinalizer ::
+    BakerId ->
+    -- |A collection of bakers and finalizers.
+    BakersAndFinalizers ->
+    -- |'True' if the consensus is part of the finalization committee.
+    -- Otherwise 'False'
+    Maybe FinalizerInfo
+isBakerFinalizer bakerId bakersAndFinalizers = do
+    -- This is O(n) but in principle we could do binary search here as the 'committeeFinalizers' are
+    -- sorted by ascending baker id.
+    Vector.find (\finalizerInfo -> finalizerBakerId finalizerInfo == bakerId) finalizers
```

Comment:
It probably doesn't matter much since we don't have huge numbers of finalizers, but wouldn't it be easier if each baker just "remembers" when going to a new payday whether they are a finalizer?

Reply:
No indeed I do not think it matters given the size of the finalization committee. But good idea @chrmatt! And GJ @td202 on that 781'er. Resolving conversation for now.